src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,520 @@
+            res(i) = loc + nn
+        end do
+    end function unif_dist_rvs_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    function unif_dist_rvs_array_${t1[0]}$${k1}$(loc, scale, array_size)        &
+        result(res)
+
+        integer, intent(in) :: array_size
+        ${t1}$, intent(in) :: loc, scale
+        ${t1}$ :: res(array_size)
+        ${t1}$ :: t
+        integer(int64) :: tmp
+        integer :: i
+
+
+        if(scale == 0._${k1}$) call error_stop("Error(unif_dist_rvs_array):"    &
+                    //" Uniform distribution scale parameter must be non-zero")
+        do i = 1, array_size
+            tmp = shiftr(dist_rand(INT_ONE), 11)
+            t = real(tmp * MESENNE_NUMBER, kind = ${k1}$)
+            res(i) = loc + scale * t
+        end do
+    end function unif_dist_rvs_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in CMPLX_KINDS_TYPES
+    function unif_dist_rvs_array_${t1[0]}$${k1}$(loc, scale, array_size)        &
+        result(res)
+
+        integer, intent(in) :: array_size
+        ${t1}$, intent(in) :: loc, scale
+        ${t1}$ :: res(array_size)
+        real(${k1}$) :: r1, tr, ti
+        integer(int64) :: tmp
+        integer :: i
+
+
+        if(scale == (0.0_${k1}$, 0.0_${k1}$)) call error_stop("Error(unif_dist_"&
+           //"rvs_array): Uniform distribution scale parameter must be non-zero")
+        do i = 1, array_size
+            tmp = shiftr(dist_rand(INT_ONE), 11)
+            r1 = real(tmp * MESENNE_NUMBER, kind = ${k1}$)
+            if(scale % re == 0.0_${k1}$) then
+                tr = loc % re
+                ti = loc % im + scale % im * r1
+            else if(scale % im == 0.0_${k1}$) then
+                tr = loc % re + scale % re * r1
+                ti = loc % im
+            else
+                tr = loc % re + scale % re * r1
+                tmp = shiftr(dist_rand(INT_ONE), 11)
+                r1 = real(tmp * MESENNE_NUMBER, kind = ${k1}$)
+                ti = loc % im + scale % im * r1
+            end if
+            res(i) = cmplx(tr, ti, kind=${k1}$)
+        end do
+    end function unif_dist_rvs_array_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in INT_KINDS_TYPES
+    elemental function unif_dist_pdf_${t1[0]}$${k1}$(x, loc, scale) result(res)
+
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+
+        if(scale == 0_${k1}$) then
+            res = 0.0
+        else if(x < loc .or. x > (loc + scale)) then
+            res = 0.0
+        else
+            res = 1. / (scale + 1_${k1}$)
+        end if
+    end function unif_dist_pdf_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in REAL_KINDS_TYPES
+    elemental function unif_dist_pdf_${t1[0]}$${k1}$(x, loc, scale) result(res)
+
+        ${t1}$, intent(in) :: x, loc, scale
+        real :: res
+
+        if(scale == 0.0_${k1}$) then
+            res = 0.0
+        else if(x <= loc .or. x >= (loc + scale)) then
```

Comment:
I just noticed that this excludes end-points, so e.g. `unif_dist_pdf(x=0.0, loc=0.0, scale=1.0) = 0`. 

Is that intended? I haven't looked extensively but thought that endpoints were usually included, for example in R the equivalent `dunif(0.0, min=0.0, max=1.0) = 1`

Reply:
Good catch! Yes, it should include end-points.